cargo-creusot/src/setup/mod.rs

```diff
@@ -0,0 +1,311 @@
+                expected_version: bin.version.to_owned(),
+            })
+        }
+    }
+
+    // check versions of the managed binaries (version in the config file vs expected version)
+    if let Config { tools: Managed { z3, cvc4, cvc5, .. } } = &config {
+        for (cur_version, bin) in [(z3, Z3), (cvc4, CVC4), (cvc5, CVC5)] {
+            if cur_version != bin.version {
+                issues.push(Issue {
+                    tool: bin.display_name.to_owned(),
+                    cur_version: Some(cur_version.to_owned()),
+                    expected_version: bin.version.to_owned(),
+                })
+            }
+        }
+    };
+
+    issues
+}
+
+fn diagnostic_extbinary(bin: ExtBinary, issues: &mut Vec<Issue>) -> anyhow::Result<PathBuf> {
+    let path = detect_binary_path(&bin).ok_or(anyhow!(
+        "{} not found. Please install {} version {}",
+        &bin.display_name,
+        &bin.display_name,
+        &bin.version
+    ))?;
+    println!("Found {} at path: {}", &bin.display_name, &path.display());
+    if let DetectedVersion::Bad(ver) = detect_binary_version(&bin, &path) {
+        issues.push(Issue {
+            tool: bin.display_name.to_owned(),
+            cur_version: ver,
+            expected_version: bin.version.to_owned(),
+        })
+    }
+    Ok(path)
+}
+
+// display the status of the creusot installation to the user
+pub fn status() -> anyhow::Result<()> {
+    let paths = get_config_paths()?;
+    match Config::read_from_file(&paths.config_file) {
+        Err(err) => {
+            println!("{err}");
+            println!(
+                "Hint: run 'cargo creusot setup install' to setup Creusot,\n\
+                      or run 'cargo creusot setup' for more information."
+            );
+        }
+        Ok(cfg) => {
+            println!("Creusot installation found.");
+            print!("{}", cfg.tools);
+            let issues = diagnostic_config(&cfg);
+            if !issues.is_empty() {
+                let severity = match cfg.tools {
+                    External { .. } => "Warning",
+                    Managed { .. } => "Error",
+                };
+                println!("");
+                for issue in &issues {
+                    println!("{severity}: {issue}")
+                }
+                if let Managed { .. } = cfg.tools {
+                    println!(
+                        "Hint: for tools installed by Creusot, \
+                              re-run 'cargo creusot setup install' \n\
+                              to upgrade them to the expected version."
+                    )
+                }
+            }
+        }
+    };
+    Ok(())
+}
+
+pub struct CreusotFlags {
+    pub why3_path: PathBuf,
+    pub why3_config: Option<PathBuf>,
+}
+
+// compute the flags to pass to creusot-rustc.
+// fail if the installation is not in an acceptable state, which means we will
+// stop there and do not attempt launching creusot-rustc.
```

Comment:
If you want these to count as documentation you can use `///` to make them doc comments.